src/WikiService.Data/RubricRepository.cs

```diff
@@ -41,31 +41,7 @@ public class RubricRepository : IRubricRepository
         dbRubric = dbRubric.OrderByDescending(rubric => rubric.CreatedAtUtc);
       }
 
-      foreach (DbRubric topRubric in dbRubric)
-      {
-        foreach (DbRubric rubric in _provider.Rubrics.AsQueryable())
-        {
-          if (rubric.ParentId == topRubric.Id)
-          {
-            topRubric.HasChild = true;
-            break;
-          }
-        }
-
-        if (!topRubric.HasChild)
-        {
-          foreach (DbArticle article in _provider.Articles.AsQueryable())
-          {
-            if (article.RubricId == topRubric.Id)
-            {
-              topRubric.HasChild = true;
-              break;
-            }
-          }
-        }
-      }
-
-      return dbRubric;
+      return FindRubricChild(dbRubric);
```

Comment:
you don't need to do it in CreateFindPredicates method, move it to FindAsync
also you can call this method after skip and take for check only neccesary objects
and change method's type to List<> or smth instead of IQueryable